### PR TITLE
renamed tmp_num_sample_values to non_na_cnt

### PR DESCRIPTION
**Pull request by zyxue**
`tmp_num_sample_values` is unclear in what it means, renamed for explicitness.

### CODE REVIEW

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for working on LightGBM!

I'm indifferent about this naming change. I'll defer to @guolinke and @shiyu1994 .

But please revert the changes to the git submodules.

image

Either you didn't branch off of `master`, in which case you should run the following

```shell
git remote add upstream https://github.com/microsoft/LightGBM.git
git checkout master
git pull upstream master
git checkout rename-tmp_num_sample_values
git merge master
```

Or you need to re-initialize the submodules, like this.

```shell
git submodule init
git submodule update --recursive
```

Please also note that development in this project is currently blocked by a serious issue in the project's continuous integration (CI). We are working on it and apologize for the inconvenience.

You can subscribe to updates on #5510 to be notified of changes.

**Review by jameslamb — COMMENTED** [review]
Thanks for the `git`  changes!

This looks ok to me, but I'll just leave a "comment" review and let @guolinke or @shiyu1994 check that this new name correctly reflects what the code is doing.

Until then... can you please merge in the latest state of `master`? We just merged a fix for the failing CI jobs. Sorry for the inconvenience.

**Review by guolinke — APPROVED** [review]
LGTM

**Pull request merged**
